File: backend/models/join_message.py
class JoinMessage:

    def __init__(self, username: str, type: str = "join_message"):
        self._username = username
        self._type = type

    def to_data(self) -> dict:
        return {
            "username": self.get_username(),
            "type": self.get_type()
        }

    def get_username(self) -> str:
        return self._username

    def get_type(self) -> str:
        return self._type

File: frontend/js/chat/models/JoinMessage.js
export default class JoinMessage {

    constructor(username, type="join_message"){
        this.username = username;
        this.type = type;
    }

    static fromData(data){
        return new JoinMessage(
            data.username,
            data.type
        )
    }

    toData(){
        return {
            username: this.username,
            type: this.type
        }
    }

    getUsername(){
        return this.username;
    }

    getType(){
        return this.type;
    }
}